src/sources/file.rs

```diff
@@ -127,13 +130,16 @@ pub struct FileConfig {
     /// By default, the [global `log_schema.host_key` option][global_host_key] is used.
     ///
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
-    #[serde(default)]
-    pub host_key: Option<OptionalValuePath>,
+    #[serde(default = "default_host_key")]
+    #[configurable(metadata(docs::examples = "hostname"))]
+    pub host_key: OptionalValuePath,
 
     /// The directory used to persist file checkpoint positions.
     ///
-    /// By default, the global `data_dir` option is used. Make sure the running user has write permissions to this directory.
+    /// By default, the global `data_dir` option is used. Make sure the running user has write
```

Comment:
Should we link to that in the global options, like host key?

Reply:
Resolved in [192c429](https://github.com/vectordotdev/vector/pull/16099/commits/192c429cac97c0da46a361108a4d292eeef97f2a)